src/node/audio_buffer_source.rs

```diff
@@ -413,6 +413,14 @@ impl AudioProcessor for AudioBufferSourceRenderer {
         // @see <https://webaudio.github.io/web-audio-api/#playback-AudioBufferSourceNode>
         let mut current_time = timestamp;
 
+        // prevent scheduling in the past
+        // If 0 is passed in for this value or if the value is less than
+        // currentTime, then the sound will start playing immediately
+        // cf. https://webaudio.github.io/web-audio-api/#dom-audioscheduledsourcenode-start-when-when
+        if !self.render_state.started && start_time < current_time {
```

Comment:
The spec also says `A RangeError exception MUST be thrown if when is negative.` but I think this is fine.

Reply:
Good catch, but I guess this should rather be checked in `start_at(when)`. 

This check will need to be done anyway: I had some very rare crash in the `many_oscillators_with_env` because `start_time` was actually lower than `current_time` (coming from some thread concurrency I guess) which completely spoiled the `phase` calculation (...that took a while to debug because I had to wait for 5 to 10mn each time... :).